JointGroupPositionController fails to initialize on Rolling due to rclcpp "undeclare_parameter" API change

There have already been [some fixes to other controllers](https://github.com/ros-controls/ros2_controllers/pull/199) for the rclcpp parameter API changes. Currently (as of 0.4.1) the `JointGroupPositionController` fails to initialize because it tries to undeclare a statically-typed parameter at [these lines](https://github.com/ros-controls/ros2_controllers/blob/6a53981982eeac1822bfca8e9d5e40156870c0b1/position_controllers/src/joint_group_position_controller.cpp#L41-L47), which produces an error:

```
[ros2_control_node-1] [INFO] [1628693935.184140389] [controller_manager]: Loading controller 'position_controller'
[ros2_control_node-1] Exception thrown during init stage with message: parameter 'interface_name' has invalid type: cannot undeclare an statically typed parameter 
[ros2_control_node-1] [ERROR] [1628693935.194137855] [controller_manager]: Could not initialize the controller named 'position_controller'
```

I wanted to get some additional context and input before I try to fix this myself: why is it necessary to undeclare the `interface_name` parameter declared by `ForwardCommandController`, and does anyone else already have a fix in mind?

--- a/effort_controllers/src/joint_group_effort_controller.cpp
+++ b/effort_controllers/src/joint_group_effort_controller.cpp
@@ -40,8 +40,10 @@ JointGroupEffortController::init(
   }
 
   try {
-    // undeclare interface parameter used in the general forward_command_controller
-    get_node()->undeclare_parameter("interface_name");
+    // Explicitly set the interface parameter declared by the forward_command_controller
+    // to match the value set in the JointGroupEffortController constructor.
+    get_node()->set_parameter(rclcpp::Parameter("interface_name", hardware_interface::HW_IF_EFFORT));
+
   } catch (const std::exception & e) {
     fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
     return controller_interface::return_type::ERROR;

--- a/effort_controllers/src/joint_group_effort_controller.cpp
+++ b/effort_controllers/src/joint_group_effort_controller.cpp
@@ -43,7 +43,7 @@ JointGroupEffortController::init(
     // Explicitly set the interface parameter declared by the forward_command_controller
     // to match the value set in the JointGroupEffortController constructor.
     get_node()->set_parameter(
-          rclcpp::Parameter("interface_name", hardware_interface::HW_IF_EFFORT));
+      rclcpp::Parameter("interface_name", hardware_interface::HW_IF_EFFORT));
   } catch (const std::exception & e) {
     fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
     return controller_interface::return_type::ERROR;

--- a/effort_controllers/src/joint_group_effort_controller.cpp
+++ b/effort_controllers/src/joint_group_effort_controller.cpp
@@ -42,8 +42,8 @@ JointGroupEffortController::init(
   try {
     // Explicitly set the interface parameter declared by the forward_command_controller
     // to match the value set in the JointGroupEffortController constructor.
-    get_node()->set_parameter(rclcpp::Parameter("interface_name", hardware_interface::HW_IF_EFFORT));
-
+    get_node()->set_parameter(
+          rclcpp::Parameter("interface_name", hardware_interface::HW_IF_EFFORT));
   } catch (const std::exception & e) {
     fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
     return controller_interface::return_type::ERROR;

--- a/position_controllers/src/joint_group_position_controller.cpp
+++ b/position_controllers/src/joint_group_position_controller.cpp
@@ -38,6 +38,16 @@ JointGroupPositionController::init(const std::string & controller_name)
     return ret;
   }
 
+  try {
+    // Explicitly set the interface parameter declared by the forward_command_controller
+    // to match the value set in the JointGroupPositionController constructor.
+    get_node()->set_parameter(rclcpp::Parameter("interface_name", hardware_interface::HW_IF_POSITION));
+
+  } catch (const std::exception & e) {
+    fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
+    return controller_interface::return_type::ERROR;
+  }
+
   return controller_interface::return_type::OK;
 }
 }  // namespace position_controllers

--- a/position_controllers/src/joint_group_position_controller.cpp
+++ b/position_controllers/src/joint_group_position_controller.cpp
@@ -38,14 +38,6 @@ JointGroupPositionController::init(const std::string & controller_name)
     return ret;
   }
 
-  try {
-    // undeclare interface parameter used in the general forward_command_controller
-    get_node()->undeclare_parameter("interface_name");
-  } catch (const std::exception & e) {
-    fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
-    return controller_interface::return_type::ERROR;
-  }
-
   return controller_interface::return_type::OK;
 }
 }  // namespace position_controllers

--- a/position_controllers/src/joint_group_position_controller.cpp
+++ b/position_controllers/src/joint_group_position_controller.cpp
@@ -41,8 +41,8 @@ JointGroupPositionController::init(const std::string & controller_name)
   try {
     // Explicitly set the interface parameter declared by the forward_command_controller
     // to match the value set in the JointGroupPositionController constructor.
-    get_node()->set_parameter(rclcpp::Parameter("interface_name", hardware_interface::HW_IF_POSITION));
-
+    get_node()->set_parameter(
+          rclcpp::Parameter("interface_name", hardware_interface::HW_IF_POSITION));
   } catch (const std::exception & e) {
     fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
     return controller_interface::return_type::ERROR;

--- a/position_controllers/src/joint_group_position_controller.cpp
+++ b/position_controllers/src/joint_group_position_controller.cpp
@@ -42,7 +42,7 @@ JointGroupPositionController::init(const std::string & controller_name)
     // Explicitly set the interface parameter declared by the forward_command_controller
     // to match the value set in the JointGroupPositionController constructor.
     get_node()->set_parameter(
-          rclcpp::Parameter("interface_name", hardware_interface::HW_IF_POSITION));
+      rclcpp::Parameter("interface_name", hardware_interface::HW_IF_POSITION));
   } catch (const std::exception & e) {
     fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
     return controller_interface::return_type::ERROR;

--- a/velocity_controllers/src/joint_group_velocity_controller.cpp
+++ b/velocity_controllers/src/joint_group_velocity_controller.cpp
@@ -40,8 +40,10 @@ JointGroupVelocityController::init(
   }
 
   try {
-    // undeclare interface parameter used in the general forward_command_controller
-    get_node()->undeclare_parameter("interface_name");
+    // Explicitly set the interface parameter declared by the forward_command_controller
+    // to match the value set in the JointGroupVelocityController constructor.
+    get_node()->set_parameter(rclcpp::Parameter("interface_name", hardware_interface::HW_IF_VELOCITY));
+
   } catch (const std::exception & e) {
     fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
     return controller_interface::return_type::ERROR;

--- a/velocity_controllers/src/joint_group_velocity_controller.cpp
+++ b/velocity_controllers/src/joint_group_velocity_controller.cpp
@@ -43,7 +43,7 @@ JointGroupVelocityController::init(
     // Explicitly set the interface parameter declared by the forward_command_controller
     // to match the value set in the JointGroupVelocityController constructor.
     get_node()->set_parameter(
-          rclcpp::Parameter("interface_name", hardware_interface::HW_IF_VELOCITY));
+      rclcpp::Parameter("interface_name", hardware_interface::HW_IF_VELOCITY));
   } catch (const std::exception & e) {
     fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
     return controller_interface::return_type::ERROR;

--- a/velocity_controllers/src/joint_group_velocity_controller.cpp
+++ b/velocity_controllers/src/joint_group_velocity_controller.cpp
@@ -42,8 +42,8 @@ JointGroupVelocityController::init(
   try {
     // Explicitly set the interface parameter declared by the forward_command_controller
     // to match the value set in the JointGroupVelocityController constructor.
-    get_node()->set_parameter(rclcpp::Parameter("interface_name", hardware_interface::HW_IF_VELOCITY));
-
+    get_node()->set_parameter(
+          rclcpp::Parameter("interface_name", hardware_interface::HW_IF_VELOCITY));
   } catch (const std::exception & e) {
     fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
     return controller_interface::return_type::ERROR;

